forte/ontology_specs/metric.json

```diff
@@ -0,0 +1,48 @@
+{ 
+    "name": "metric", 
+    "definitions": [ 
+      {
+        "entry_name": "ft.onto.metric.BaseMetric", 
+        "parent_entry": "forte.data.ontology.top.Generics", 
+        "description": "A base metric entity, all metric entities should inherit from it.", 
+        "attributes": [ 
+          { 
+              "name": "metric_name", 
+              "type": "str" 
+          }
+        ]
+      },
+      { 
+        "entry_name": "ft.onto.metric.SingleMetric", 
+        "parent_entry": "ft.onto.metric.BaseMetric", 
+        "description": "A single metric entity, used to present a metric of one float (e.g. accuracy).", 
+        "attributes": [ 
+          { 
+              "name": "value", 
+              "type": "float" 
+          }
+        ] 
+      }, 
+      { 
+        "entry_name": "ft.onto.metric.NdMetric", 
+        "parent_entry": "ft.onto.metric.BaseMetric", 
+        "description": "A N-dimensional metric entity, used to present a metric of N-d array (e.g. confusion matrix).", 
+        "attributes": [ 
+          { 
+              "name": "value", 
+              "type": "NdArray",
+              "value_type": "numpy.ndarray"
+          },
+          {
```

Comment:
attributes like `shape` and `dtype` should be part of `NdArray` property, they shouldn't be separate attributes to handle. And `"value_type": "numpy.ndarray"` looks redundant to me, we surely cannot support any arbitrary value type here so we either hard-code it, or only provide a few implementations for user to choose from, we should not use arbitrary class.